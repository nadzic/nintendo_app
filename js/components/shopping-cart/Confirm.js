import React, { Component } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { get as safeGet } from 'lodash';
import { COLORS } from 'nintendoapp/js/constants/colors';
import { calculateTotal } from 'nintendoapp/js/utils/calculation';


/* Config/Constants
============================================================================= */

/* eslint-disable global-require */

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

type Props = {
  navigation: any,
  ui: any
};

class Confirm extends Component<Props> {
  // TODO: disconnect from store
  goToNextScreen() {
    const {
      navigation,
    } = this.props;

    navigation.navigate('Payment');
  }

  render() {
    const products = safeGet(this.props, 'products', []);

    const numProducts = products.length;
    const total = calculateTotal(products);
    return (
      <View style={[styles.container, numProducts < 1 && styles.disabled]}>
        <View style={styles.totalSection}>
          <Text style={styles.totalText}>Total : </Text>
          <Text style={styles.valueText}>${total}</Text>
        </View>
        <TouchableOpacity
          style={styles.confirmButton}
          onPress={() => this.goToNextScreen()}
          disabled={numProducts < 1}
        >
          <Text style={styles.confirmText}>Confirm and Pay</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: COLORS.LIGHTGREY3,
    position: 'absolute',
    width: '100%',
    bottom: 0,
    borderTopColor: COLORS.LIGHTGREY,
    borderTopWidth: 0.5,
    paddingTop: 7,
  },
  disabled: {
    opacity: 0.3,
  },
  totalSection: {
    display: 'flex',
    flexDirection: 'row',
  },
  totalText: {
    color: COLORS.BLACK,
    fontSize: 18,
    marginBottom: 8,
    fontFamily: 'BrandonText-Bold',
  },
  valueText: {
    color: COLORS.GREEN,
    fontSize: 18,
    fontFamily: 'BrandonText-Bold',
  },
  confirmButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginBottom: 15,
  },
  confirmText: {
    fontFamily: 'BrandonText-Medium',
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

export default Confirm;
