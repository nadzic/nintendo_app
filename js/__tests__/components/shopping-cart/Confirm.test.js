import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Confirm from 'nintendoapp/js/components/shopping-cart/Confirm';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
  };

  const wrapper = shallow(<Confirm {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('Confirm component', () => {
  it('should renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
