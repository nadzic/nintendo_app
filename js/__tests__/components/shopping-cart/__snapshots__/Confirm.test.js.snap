// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm component should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f3f3f3",
        "borderTopColor": "#d1d1d1",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "display": "flex",
        "paddingTop": 7,
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "opacity": 0.3,
      },
    ]
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "BrandonText-Bold",
          "fontSize": 18,
          "marginBottom": 8,
        }
      }
    >
      Total : 
    </Text>
    <Text
      style={
        Object {
          "color": "#0EAF27",
          "fontFamily": "BrandonText-Bold",
          "fontSize": 18,
        }
      }
    >
      $
      0
    </Text>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={true}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0EAF27",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 15,
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 20,
        }
      }
    >
      Confirm and Pay
    </Text>
  </TouchableOpacity>
</View>
`;
